docs/data-structures.md

```diff
@@ -0,0 +1,143 @@
+metadata is meant to live, so that our internal schema can grow safely, without
+risk of colliding with attributes required by the system.
+
+We use Protocol Labs' `multihash` self-describing hashing scheme for hashing the
+contents of our objects. IPFS uses an object's `multihash` as its address within
+the system--all objects are retrievable by their hash. IPFS also features a
+built in faculty for linking objects by their content-addresses called IPLD.
+Under IPLD, IPFS users may express traversals over arbitrary graphs of linked
+objects via a pathing markup very similar to that of HTTP. Links are encoded
+using a special property name/sub-object structure:
+
+```
+{myLink: {@link: <reference>}}
+```
+
+If this object had a hash of <foobar>, one could access the object it linkes to
+by looking up the path `/ipfs/<foobar>/myLink`. Following this, all instances of
+`<Reference>` in our first RFC correspond to objects of this structure.
+
+### Entities
+
+Entities bear a nearly identical structure:
+
+```
+Entity = {
+ "type" : "entity"
+ ["keychain" : <Reference>]
+ "signatures" : <Signatures>
+ "meta" : {
+  "name" : <String>
+  <Key> : <Value> ... ; entity metadata
+  }
+ ["metaSource": <Reference>]
+ }
+```
+
+The only discrepancy between the two would be the expected `meta` fields,
+"created" and "description".
+
+## Chain Structures
+
+Mediachain is comprised of an unbounded number of chains--reverse linked
+lists--that represent the complete history of revisions of entities and
+artefacts managed by the Mediachain. There are are two classes of chains, the
+system-wide blockchain and the individual entity/artefact revision chains.
+
+### Entity/Artefact Chains
+
+When a new object is inserted into Mediachain for the first time, an object
+following the schema detailed above is written to IPFS. The address of that
+initial submission is what is referred to as its canonical identifier--this will
+never change. All revisions to and extensions of this metadata are then written
+into a distinct chain, indexed by this canonical identifier. The updates in this
+chain are serialized via the consensus layer. Each update is signed by its
+author, so that users can ignore updates authored by peers they don't know or
+trust.
+
+The heads of these individual chains are stored in the index, which is
+serialized out to a journal much like that of a filesystem. Maintaining and
+tracking this journal is the responsibility of the transactors. More on the
+journal later!
+
+### Blockchain
+
+Whereas the individual chains in Mediachain are somewhat unique to its design,
+Mediachain also features a blockchain that stores batches of updates in blocks
+as one would expect. The purpose of this blockchain is to facilitate a simpler,
+more efficient "catch-up" for new nodes joining the system. By adjusting the
+block size to include multiple updates per block, applications wishing to ingest
+the Mediachain history can do so with fewer queries.
+
+## The Journal
```

Comment:
I think the journal/blockchain part needs a bit more love, it's the most unclear (and important) aspect of the system. @vyzo can you comment on this?